### PR TITLE
fix: Prevent duplicate EPUB/DPUB footnote body generation for repeated noteref targets

### LINKED ISSUE #1767: EPUB/DPUB footnote duplication
Test case: [dpub-footnote-duplicate.html](https://gist.githubusercontent.com/MurakamiShinyu/901f1aae86fc69bb7680b64608fbed59/raw/dpub-footnote-duplicate.html)

```html
<p>Aaa<a id="fnref1" href="#fn1" role="doc-noteref"><sup>1</sup></a>
   bbb<a id="fnref2" href="#fn2" role="doc-noteref"><sup>2</sup></a>.</p>

<p>Ccc aaa<a id="fnref1-1" href="#fn1" role="doc-noteref"><sup>1</sup></a>.</p>

<aside id="fn1" role="doc-footnote">
  <a href="#fnref1" role="doc-backlink"><sup>1</sup></a>Aaaaaa
</aside>
<aside id="fn2" role="doc-footnote">
  <a href="#fnref2" role="doc-backlink"><sup>2</sup></a>Bbbbbb
</aside>
```

Test with Vivliostyle Viewer (canary): https://vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/901f1aae86fc69bb7680b64608fbed59/raw/dpub-footnote-duplicate.html

Result screenshot:

Image

The problem is that the footnote "1 Aaaaaa" appears twice. When there are multiple footnote reference elements with the same `href="#fn1", the referenced footnote (id="fn1") should not be duplicated.

### PR DESCRIPTION
**Pull request by MurakamiShinyu**
- render semantic footnote body only for the first noteref in source order per target
- avoid retry-sensitive state that could suppress footnote rendering
- add regression tests for both DPUB role-based and EPUB type-based duplicate noteref cases
- fixes #1767

Assisted-by: GitHub Copilot Chat

<details>
<summary>How the AI was used</summary>

- The human author asked the AI to fix issue #1767 (duplicate footnote bodies for repeated noterefs referencing the same footnote); across several rounds, one attempt initially broke footnote rendering entirely, which was diagnosed and fixed using the Chrome DevTools MCP for verification.
- The human author asked for regression tests to be registered in `file-list.js` with self-descriptive titles, questioned an apparently redundant DOM scan the AI had added (leading to a simplification using a shared helper), and asked for a `@fileoverview`-style header comment matching repository conventions.
- After this fix was committed, the human author asked the AI to prepare a separate follow-up commit bundling an unrelated internal-link-navigation fix with a footnote-backlink underline style change the human author had made directly, framed as a follow-up to the (already-closed) issue #1767.

</details>

### CODE REVIEW

File: packages/core/src/vivliostyle/vgen.ts [hunk 1]
```diff
@@ -159,9 +163,60 @@ export class ViewFactory
...
+          anchor.getAttributeNS(Base.NS.XLINK, "href");
+        if (!anchorHref || anchorHref.startsWith("javascript:")) {
+          continue;
+        }
+        if (Base.resolveURL(anchorHref, this.xmldoc.url) === resolvedHref) {
+          firstOffset = this.xmldoc.getElementOffset(anchor);
+          break;
+        }
+      }
+      this.semanticFootnoteFirstRefOffsets.set(resolvedHref, firstOffset);
```
[the remark this answers is not in the document]
Comment (MurakamiShinyu) [reply packages/core/src/vivliostyle/vgen.ts]: Fixed in [c9f86b9](https://github.com/vivliostyle/vivliostyle.js/pull/1768/commits/c9f86b9eeeb2e106cfe60d8f3c8aa59a6dffbb6d)
